src/Compilers/VisualBasic/Portable/Symbols/Source/OverloadingHelper.vb

```diff
@@ -20,15 +24,13 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
         ''' The rules are as follows:
         '''    1) If a method or property overrides one from its base class, its metadata name
         '''       must match that.
-        '''    2) If method overload those in the base (because the Overloads keyword is used), and
+        '''    2) If method overloads those in the base (because the Overloads keyword is used), and
         '''       all metadata names in the base are consistent in case, use that name.
         '''    3) All overloads with a class should match, except possibly for overrides. If there is
         '''       an override or overload from base, use that. Otherwise, use casing of first member in
         '''       class.
         ''' </summary>
-        Public Sub SetMetadataNameForAllOverloads(name As String, kind As SymbolKind, container As NamedTypeSymbol)
-            Debug.Assert(kind = SymbolKind.Method OrElse kind = SymbolKind.Property)
-
+        Private Sub SetMetadataNameForAllOverloads(name As String, container As NamedTypeSymbol, predicate As Predicate(Of Symbol))
```

Comment:
Rather than introducing a predicate, could we simply add a check for ```Not member,IsAccessor()``` where the kind is checked? It looks like that would work for both methods and properties.